keycloakClient/keycloakClient.go

```diff
@@ -104,16 +119,10 @@ func (c *Client) AutoRefreshToken(realm string, username string, password string
 		onFailure(err)
 		return
 	}
-	// Refresh before the auth token expires
-	nextRefresh := info.Expires.Sub(time.Now().Add(time.Duration(c.refreshAuthTokenBeforeExpiration) * time.Second))
+	info.autorefreshes = true
+	info.onRefreshFailure = onFailure
 	// Pass in arguments to allow original args to get garbage collected.
-	info.refresher = time.AfterFunc(nextRefresh, func(realm, username, password string, onFailure func(error), c *Client) func() {
-		// send back a function which will re-call this method after the timeout
-		// capturing the arguments in a closure.
-		return func() {
-			c.AutoRefreshToken(realm, username, password, onFailure)
-		}
-	}(realm, username, password, onFailure, c))
```

Comment:
i extracted this to a first order func that could also be used by `GetTokenInfo` which is responsible for creating a new token on refresh.